hail/src/main/scala/is/hail/expr/ir/StringTableReader.scala

```diff
@@ -34,6 +39,78 @@ object StringTableReader {
     status
   }
 }
+case class StringTablePartitionReader(lines: GenericLines) extends PartitionReader{
+  override def contextType: Type = lines.contextType
+
+  override def fullRowType: Type = TStruct("file"-> TString, "text"-> TString)
+
+  override def rowRequiredness(requestedType: Type): TypeWithRequiredness = {
+    val req = BaseTypeWithRequiredness.apply(requestedType).asInstanceOf[RStruct]
+    req.fields.foreach(field => field.typ.hardSetRequiredness(true))
+    req.hardSetRequiredness(true)
+    req
+  }
+
+  override def emitStream(
+     ctx: ExecuteContext,
+     cb: EmitCodeBuilder,
+     context: EmitCode,
+     partitionRegion: Value[Region],
+     requestedType: Type): IEmitCode = {
+
+     context.toI(cb).map(cb) { partitionContext =>
+       val iter = cb.emb.genFieldThisRef[CloseableIterator[GenericLine]]("iter")
+
+       val ctxMemo = partitionContext.asBaseStruct.memoize(cb, "string_table_reader_ctx")
+       val fileName = cb.emb.genFieldThisRef[String]("fileName")
+       val line = cb.emb.genFieldThisRef[String]("line")
+
+       SStreamCode(new StreamProducer {
+         override val length: Option[EmitCodeBuilder => Code[Int]] = None
+
+         override def initialize(cb: EmitCodeBuilder): Unit = {
+           val contextAsJavaValue = coerce[Any](StringFunctions.scodeToJavaValue(cb, partitionRegion, ctxMemo))
+
+           cb.assign(fileName, ctxMemo.loadField(cb, "file").get(cb).asString.loadString())
+
+           cb.assign(iter,
+             cb.emb.getObject[(Any) => CloseableIterator[GenericLine]](lines.body)
+               .invoke[Any, CloseableIterator[GenericLine]]("apply", contextAsJavaValue)
+           )
+         }
+
+         override val elementRegion: Settable[Region] =
+           cb.emb.genFieldThisRef[Region]("string_table_reader_region")
+
+         override val requiresMemoryManagementPerElement: Boolean = true
+
+         override val LproduceElement: CodeLabel = cb.emb.defineAndImplementLabel { cb =>
+           val hasNext = iter.invoke[Boolean]("hasNext")
+           cb.ifx(hasNext, {
+             val gLine = iter.invoke[GenericLine]("next")
+             cb.assign(line, gLine.invoke[String]("toString"))
+             cb.goto(LproduceElementDone)
+           }, {
+             cb.goto(LendOfStream)
+           })
+         }
+         override val element: EmitCode = EmitCode.fromI(cb.emb) { cb =>
+           val reqType: TStruct = requestedType.asInstanceOf[TStruct]
+
+           IEmitCode.present(cb, SStackStruct.constructFromArgs(cb, elementRegion, reqType,
+             EmitCode.fromI(cb.emb)(cb => IEmitCode.present(cb, SJavaString.construct(fileName))),
+             EmitCode.fromI(cb.emb)(cb => IEmitCode.present(cb, SJavaString.construct(line)))))
+         }
```

Comment:
You are getting errors, I think it is because you need to use the `requestedType` when you do this and only pass in arguments that are requested. You are currently always passing in `fileName` and `line`. Think back to the old method where we only did `RegionValueBuilder.addAnnotation` if the field was in the requested type. Same idea here